src/SurfaceEnergyFluxVegetatedMod.F90

```diff
@@ -305,14 +305,14 @@ subroutine SurfaceEnergyFluxVegetated(noahmp)
        EnergyResTmp         = RadSwAbsVeg - RadLwNetCanopy - HeatSensibleCanopy - &
                               HeatLatentCanEvap - HeatLatentCanTransp + HeatPrecipAdvCanopy
        FluxTotCoeff         = VegFrac * (4.0*LwCoeffCan*TemperatureCanopy**3 + ShCoeff + &
-                                        (LhCoeff+TranspHeatCoeff)*VapPresSatCanTempD) + HeatCapacCan/MainTimeStep     ! volumetric heat capacity
+                                        (LhCoeff+TranspHeatCoeff)*VapPresSatCanTempD + HeatCapacCan/MainTimeStep)     ! volumetric heat capacity
        TemperatureCanChg    = EnergyResTmp / FluxTotCoeff
        ! update fluxes with temperature change
        RadLwNetCanopy       = RadLwNetCanopy + VegFrac * 4.0 * LwCoeffCan * TemperatureCanopy**3 * TemperatureCanChg
        HeatSensibleCanopy   = HeatSensibleCanopy + VegFrac * ShCoeff * TemperatureCanChg
        HeatLatentCanEvap    = HeatLatentCanEvap + VegFrac * LhCoeff * VapPresSatCanTempD * TemperatureCanChg
        HeatLatentCanTransp  = HeatLatentCanTransp  + VegFrac * TranspHeatCoeff * VapPresSatCanTempD * TemperatureCanChg
-       HeatCanStorageChg    = TemperatureCanChg * HeatCapacCan / MainTimeStep   ! canopy heat storage change [W/m2]
+       HeatCanStorageChg    = TemperatureCanChg * (VegFrac * HeatCapacCan / MainTimeStep)   ! canopy heat storage change [W/m2]
```

Comment:
Good suggestion. I will re-arrange the terms.

Reply:
Hi,
I am reviewing and validating the code changes. I agree with the code changes, but have questions about results for validation.
I am running a real case in a site with  DYNAMIC_VEG_OPTION = 4.
I perform two simulations, one is maximum vegetation fraction of 0.96, the other is set to 0.1.

May I suppose getting the same vegetation temperature for two simulations as shown by https://github.com/NCAR/noahmp/issues/86#issue-1737337322

Thanks.

Reply:
@tslin2 I should have put more details in the issue, but this has been evolving a bit as @cenlinhe and I have talked over the last few days. You should get the same vegetation temperature under some strict conditions for now:
- no precipitation (there is a separate issue with fwet)
- no solar radiation (there is a separate issue with radiation)
- only on the first iteration of vege_flux (old code naming convention) since with the combined soil over bare and vegetation tiles, the tveg will evolve differently because the soil temperature will be different

Reply:
@barlage Thanks for your quick response. This make sense to me, and I agree because solar radiation and others also change by vegetation fraction, so it cannot be validated in a real case simulation.
Thanks all.